setup.py

```diff
@@ -1,15 +1,80 @@
 #!/usr/bin/env python
+import os
+import subprocess
+import sys
+from glob import glob
+from setuptools import setup, find_packages, Extension
 
-try:
-    from setuptools import setup
-except ImportError:
-    from distribute_setup import use_setuptools
-    use_setuptools()
-    from setuptools import setup
+
+if os.path.exists('relic'):
+    sys.path.insert(1, 'relic')
+    import relic.release
+else:
+    try:
+        import relic.release
+    except ImportError:
+        try:
+            subprocess.check_call(['git', 'clone',
+                'https://github.com/jhunkeler/relic.git'])
+            sys.path.insert(1, 'relic')
+            import relic.release
+        except subprocess.CalledProcessError as e:
+            print(e)
+            exit(1)
+
+
+version = relic.release.get_info()
+relic.release.write_template(version, 'lib/reftools')
 
 setup(
-    setup_requires=['d2to1>=0.2.9', 'stsci.distutils>=0.3.2'],
-    d2to1=True,
-    use_2to3=False,
-    zip_safe=False
+    name = 'reftools',
+    version = version.pep386,
```

Comment:
No, the tag for 1.7 would be there because that's part of the development cycle. No tag = no data. 

You need to tag the repo 2.0 when the _code_ is to be considered 2.0. Until then it's "1.7 + development toward another goal"

**Current model:**
Hardcoded `VERSION="2.0.dev"`
Post-package version data: `reftools-2.0.dev[count_of_all_commits]`

RELIC model:
Softcode `VERSION=version.pep386`
Post-package version data: `reftools-1.7.dev[count_of_commits_past_tag]`

RELIC model (cont.)

Assuming code in development is ready for release as 2.0:
`% git tag -a 2.0 -m "Big 'ole changes" && git push --tags`

The next time this package is compiled, the version is reported as:
`reftools-2.0`

`python setup.py sdist` dumps `reftools-2.0.tar.gz`

Additional development reports `reftools-2.0.dev10` after compiling.

Tagging 2.1 reports `reftools-2.1` after compiling and so on.